### PR TITLE
Update per-reporter staleness schema to be more precise

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [ESSNTL-1032](https://issues.redhat.com/browse/ESSNTL-1032).
It updates the API schema for the per-reporter staleness field to be more precise. Instead of just defining a general object, it specifies the dictionary structure along with the 3 fields in the "value" part of the key-value pair.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: swagger/api.spec.yaml [hunk 1]
```diff
@@ -982,6 +982,17 @@ components:
               description: >-
                 Reporting source of the last checkin status, stale_timestamp, and last_check_in.
               type: object
+              additionalProperties:
```
Comment (RonnyPfannschmidt) [inline swagger/api.spec.yaml:985]: additionalProperties is incorrect ,please use properties, and create a own schema entry to support the code genrator

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]

File: swagger/api.spec.yaml (hunk 1 above)
Comment (kruai) [reply swagger/api.spec.yaml:985]: OK, I can make it work that way

Comment (kruai) [reply swagger/api.spec.yaml:985]: properties works differently from additionalProperties, though, right? I was using additionalProperties to make it a dictionary, key-value pair style

**Review by thearifismail — DISMISSED** [review]
LGTM

File: swagger/api.spec.yaml (hunk 1 above)
Comment (RonnyPfannschmidt) [reply swagger/api.spec.yaml:985]: my mistake when reading, the levelling is correct, i wish there was a map type

Comment (kruai) [reply swagger/api.spec.yaml:985]: Same! Thanks for checking, though.

File: swagger/api.spec.yaml [hunk 2]
```diff
@@ -1060,7 +1062,17 @@ components:
             name.
           type: string
           example: host1.mydomain.com
-
+    PerReporterStaleness:
+      type: object
+      properties:
+        last_check_in:
```
Comment (ryandillinfelton) [inline swagger/api.spec.yaml:1068]: Can you add an example value here, and for `stale_timestamp`

Comment (kruai) [reply swagger/api.spec.yaml:1068]: Done!

**Review by ryandillinfelton — APPROVED** [review]
Thank you!

File: swagger/api.spec.yaml [hunk 3]
```diff
@@ -1060,7 +1062,19 @@ components:
...
+          type: string
+          format: date-time
+          example: '2020-02-10T08:07:03.354307+00:00'
+        stale_timestamp:
```
Comment (RonnyPfannschmidt) [inline swagger/api.spec.yaml:1072]: an optional follow-up would be to make a shared item for datetime formats

**Review by RonnyPfannschmidt — APPROVED** [review]

**Review by RonnyPfannschmidt — APPROVED** [review]

**Review by thearifismail — COMMENTED** [review]
There should be a test using zero timezone, i.e. timestamp ending with "Z" e.g. 2020-02-13T12:08:55Z

**Comment by kruai** [comment]
> There should be a test using zero timezone, i.e. timestamp ending with "Z" e.g. 2020-02-13T12:08:55Z

What kind of test do you mean? We don't have unit tests for the swagger doc. And we do have unit tests for datetime formats in my other PR: https://github.com/RedHatInsights/insights-host-inventory/pull/988

**Comment by thearifismail** [comment]
> > There should be a test using zero timezone, i.e. timestamp ending with "Z" e.g. 2020-02-13T12:08:55Z
> 
> What kind of test do you mean? We don't have unit tests for the swagger doc. And we do have unit tests for datetime formats in my other PR: #988

Yes I was looking for using this format "2021-05-11T19:46:56.951+00:00Z" in `per_reporter_staleness`.  Since the format is being tested in PR #988, it should be good.

**Review by thearifismail — APPROVED** [review]

**Pull request merged**